bson/src/main/org/bson/internal/ProvidersCodecRegistry.java

```diff
@@ -51,7 +51,7 @@ public ProvidersCodecRegistry(final List<? extends CodecProvider> codecProviders
     }
 
     @SuppressWarnings({"rawtypes" })
-    public <T> Codec<T> get(final ChildCodecRegistry<T> context) {
+    public synchronized <T> Codec<T> get(final ChildCodecRegistry<T> context) {
```

Comment:
When running a simple multi threaded program there is a race condition when creating automatic pojos.  The following program highlights the registry gets and creates multiple Codecs for `SimpleModel.class`.  

```
ExecutorService executor = Executors.newFixedThreadPool(100);
final CodecRegistry codecRegistry = fromProviders(asList(new ValueCodecProvider(), PojoCodecProvider.builder().automatic(true).build()));
List<Callable<String>> callables = Stream.iterate(0, n -> n + 1).limit(1000)
        .map( i -> (Callable<String>) () -> codecRegistry.get(SimpleModel.class).toString())
        .collect(Collectors.toList());
try {
    executor.invokeAll(callables)
            .stream()
            .map(future -> {
                try {
                    return future.get();
                }
                catch (Exception e) {
                    throw new IllegalStateException(e);
                }
            })
            .forEach(System.out::println);
} catch (InterruptedException e) {
    e.printStackTrace();
}
```

By synchronizing here we can stop multiple creations of the `AutomaticPojoCodec` for a single class.